src/main/java/org/zalando/nakadi/service/subscription/SubscriptionTimeLagService.java

```diff
@@ -82,21 +83,12 @@ public SubscriptionTimeLagService(final TimelineService timelineService,
             for (final EventTypePartition partition : futureTimeLags.keySet()) {
                 timeLags.put(partition, futureTimeLags.get(partition).get());
             }
-            return timeLags;
-        } catch (final RejectedExecutionException e) {
-            throw new LimitReachedException("Time lag statistics thread pool exhausted", e);
-        } catch (final TimeoutException e) {
-            throw new TimeLagStatsTimeoutException("Timeout exceeded for time lag statistics", e);
-        } catch (final ExecutionException e) {
-            if (e.getCause() instanceof NakadiBaseException) {
-                throw (NakadiBaseException) e.getCause();
-            } else {
-                throw new InconsistentStateException("Unexpected error occurred when getting subscription time lag",
-                        e.getCause());
-            }
-        } catch (final Throwable e) {
-            throw new InconsistentStateException("Unexpected error occurred when getting subscription time lag", e);
+        } catch (RejectedExecutionException|TimeoutException | ExecutionException  e) {
+            LOG.warn("caught exception the timelag stats are not complete");
+        } catch (Throwable e){
```

Comment:
@Kunal-Jha I'm afraid that this might introduce some problems. Imagine that Kafka is unavailable, then we should probably bubble up this exception. I would say that we should only "silence" the exceptions related to timeout problem when collecting time lag.